tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -17,35 +18,1261 @@
+		return errno;
+	}
+
+	if (octets_per_frame < 0) {
+		LOG_DBG("Error: Invalid octets per frame.");
+		return errno;
+	}
+
+	if (frames_per_sdu < 0) {
+		LOG_DBG("Error: Invalid frames per sdu.");
+		return errno;
+	}
+
+	if (chan_allocation_err < 0) {
+		LOG_DBG("Error: Invalid channel allocation.");
+		return errno;
+	}
+
+	return 0;
+}
+
+static int lc3_config(struct bt_conn *conn, const struct bt_bap_ep *ep, enum bt_audio_dir dir,
+		      const struct bt_codec *codec, struct bt_bap_stream **stream,
+		      struct bt_codec_qos_pref *const pref, struct bt_bap_ascs_rsp *rsp)
+{
+	struct audio_connection *audio_conn;
+	struct audio_stream *stream_wrap;
+
+	LOG_DBG("ASE Codec Config: ep %p dir %u", ep, dir);
+
+	print_codec(codec);
+
+	if (validate_codec_parameters(codec)) {
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_CONF_REJECTED,
+				       BT_BAP_ASCS_REASON_CODEC_DATA);
+
+		btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP,
+						     BTP_ASCS_STATUS_FAILED);
+
+		return -ENOTSUP;
+	}
+
+	audio_conn = find_connection(conn);
+	stream_wrap = stream_alloc(audio_conn);
+
+	if (stream_wrap == NULL) {
+		LOG_DBG("No free stream available");
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_NO_MEM, BT_BAP_ASCS_REASON_NONE);
+
+		btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP,
+						     BTP_ASCS_STATUS_FAILED);
+
+		return -ENOMEM;
+	}
+
+	*stream = &stream_wrap->stream;
+	LOG_DBG("ASE Codec Config stream %p", *stream);
+
+	if (dir == BT_AUDIO_DIR_SOURCE) {
+		audio_conn->configured_source_stream_count++;
+	} else {
+		audio_conn->configured_sink_stream_count++;
+	}
+
+	*pref = qos_pref;
+
+	return 0;
+}
+
+static int lc3_reconfig(struct bt_bap_stream *stream, enum bt_audio_dir dir,
+			const struct bt_codec *codec, struct bt_codec_qos_pref *const pref,
+			struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("ASE Codec Reconfig: stream %p", stream);
+
+	print_codec(codec);
+
+	return 0;
+}
+
+static int lc3_qos(struct bt_bap_stream *stream, const struct bt_codec_qos *qos,
+		   struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("QoS: stream %p qos %p", stream, qos);
+
+	print_qos(qos);
+
+	return 0;
+}
+
+static int lc3_enable(struct bt_bap_stream *stream, const struct bt_codec_data *meta,
+		      size_t meta_count, struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("Enable: stream %p meta_count %zu", stream, meta_count);
+
+	return 0;
+}
+
+static int lc3_start(struct bt_bap_stream *stream, struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("Start: stream %p", stream);
+
+	return 0;
+}
+
+static bool valid_metadata_type(uint8_t type, uint8_t len, const uint8_t *data)
+{
+	switch (type) {
+	case BT_AUDIO_METADATA_TYPE_PREF_CONTEXT:
+	case BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT:
+		if (len != 2) {
+			return false;
+		}
+
+		/* PTS wants us to reject the parameter if reserved bits are set */
+		if ((*(uint16_t *)data & ~(uint16_t)BT_AUDIO_CONTEXT_TYPE_ANY) > 0) {
+			return false;
+		}
```

Comment:
@sjanc is the BT tester always a little-endian device? 

Since the data is always LE, we either need to ensure that the device running this test is also LE, or we need to use `sys_get_le16`.